src/main/app/controller/courts/LocalAuthoritiesController.ts

```diff
@@ -0,0 +1,168 @@
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    error = '',
+    courtLocalAuthorities: LocalAuthority[] = null): Promise<void> {
+    const slug: string = req.params.slug as string;
+    const areaOfLaw: string = req.params.areaOfLaw as string;
+
+    let areasOfLaw: AreaOfLaw[] = null;
+    await req.scope.cradle.api.getCourtAreasOfLaw(slug)
+      .then((value: AreaOfLaw[]) => areasOfLaw = this.checkFamilyAreasOfLaw(value))
+      .catch(() => {
+        error += this.getCourtAreasOfLawErrorMsg;
+      } );
+
+    if ( courtLocalAuthorities === null  || (courtLocalAuthorities.length < 1 && areaOfLaw != 'unknown')) {
+
+      await req.scope.cradle.api.getCourtLocalAuthoritiesByAreaOfLaw(slug, areaOfLaw)
+        .then((value: LocalAuthority[]) => courtLocalAuthorities = value)
+        .catch(() => error += this.getCourtLocalAuthoritiesErrorMsg);
+    }
+
+    let allLocalAuthorities: LocalAuthority[] = [];
+    await req.scope.cradle.api.getLocalAuthorities()
+      .then((value: LocalAuthority[]) => allLocalAuthorities = value)
+      .catch(() => error += this.getLocalAuthoritiesErrorMsg);
+
+    const pageData: LocalAuthoritiesPageData = {
+      errorMsg: error,
+      updated: updated,
+      selectedAreaOfLaw: areaOfLaw,
+      courtAreasOfLaw: this.getAreasOfLawForSelect(areasOfLaw, areaOfLaw),
+      localAuthoritiesItems: this.mapLocalAuthorityToLocalAuthorityItem(allLocalAuthorities, courtLocalAuthorities)
+    };
+
+    res.render('courts/tabs/localAuthoritiesContent', pageData);
+
+  }
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+
+    let localAuthorities: LocalAuthority[] = [];
+
+    if (req.body.localAuthoritiesItems){
+      localAuthorities = Array.isArray(req.body.localAuthoritiesItems) ? req.body.localAuthoritiesItems.map((la: string) => JSON.parse(la)) : [JSON.parse(req.body.localAuthoritiesItems)];
+    }
+
+    if(!CSRF.verify(req.body._csrf)) {
+      return this.getLocalAuthorities(req, res, false, this.updateErrorMsg, localAuthorities);
+    }
+    else
+    {
+      await req.scope.cradle.api.updateCourtLocalAuthoritiesByAreaOfLaw(req.params.slug, req.params.areaOfLaw, localAuthorities)
+        .then((value: LocalAuthority[]) => this.getLocalAuthorities(req, res, true, '', value))
+        .catch(() => this.getLocalAuthorities(req, res, false, this.updateErrorMsg, localAuthorities));
+    }
+
+  }
+
+  private mapLocalAuthorityToLocalAuthorityItem(allLocalAuthorities: LocalAuthority[], courtLocalAuthorities: LocalAuthority[]): LocalAuthorityItem[] {
+
+    if(courtLocalAuthorities) {
+
+      const localAuthorityItems = allLocalAuthorities.map((la: LocalAuthority) => (
+        {
+          value: JSON.stringify(la),
+          text: la.name,
+          checked: courtLocalAuthorities.some(e => e.id === la.id)
+        }));
+
+      return localAuthorityItems.sort((a, b) => (a.text < b.text ? -1 : 1));
+    }
+    else
+    {
+      return [];
```

Comment:
Is this an error scenario? If so, is it worth adding in a log error here? Not a problem if it's default behaviour 

If it's default though, can always do something like: 

```
    if(courtLocalAuthorities) {
}

> if we find this is an error case, which it doesn't look like it is - log.error('this was not expected') 
return [];
```

Reply:
its default behaviour